### PR TITLE
Buff Treant

### PR DESCRIPTION
**Pull request by TheUnholyWrath**
I think treant is the worst hero in the game right now

- Readd Eyes in the Forest + Overgrowth AoE talent
- Shuffle other talents with buffed/nerfed values
- Rescale Leech Seed and Living Armor 
- Buff Nature's Grasp OAA levels 

Here's my proposed talent tree:

25  450 AoE Living Armor             |  +350 Eyes in the Forest + Overgrowth AoE
20  +80 Leech Seed Damage        | +8 Living Armor Bonus Armor
15  +75 DPS Overgrowth              | +20% Leech Seed Movement Slow
10  -5s Nature's Grasp Cooldown | +30 Nature's Grasp Damage

Right now treant is maybe map vision and sometimes giving 1 person armor. Losing the eyes in the forest aoe talent made his vision inconsistent and awkward. The damage values on his abilites are all 0. 

750 Healing Living Armor at level 4 is probably the worst change I've made. I was going to make a level 20 talent for it with like +500 heal but it sat across from +8 armor and I like that talent where it is (could just add in 8 armor to the ability and replace it with a heal talent). I think its a pretty powerful thing to put on your carry but 1) you are a treant with 4 points in living armor 2) its dispellable 3) by the time you can get to level 7 I'm not certain that its broken. 

Leech Seed is probably the worst ability in the whole game. It should probably scale in duration. When I end up nerfing level 4 living armor I'll add it in.

### CODE REVIEW

File: game/scripts/npc/abilities/treant_natures_grasp.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@
       "01"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "damage_per_second"                               "30 50 70 90 180 360" //OAA
+        "damage_per_second"                               "30 50 70 90 270 450" //OAA
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/treant_natures_grasp.txt:40]: I am not a fan of this kind of buffing. May I suggest the scaling like this: 30 60 90 120 240 480

File: game/scripts/npc/abilities/treant_living_armor.txt [hunk 2]
```diff
@@ -35,7 +35,7 @@
       "01"
       {
         "var_type"                                        "FIELD_INTEGER"
-        "total_heal"                                      "180 270 360 450 900 1800" //OAA
+        "total_heal"                                      "150 350 550 750 1500 3000" //OAA
```
Comment (DarkoniusXNG) [inline game/scripts/npc/abilities/treant_living_armor.txt:38]: You are aware that 750 is too much. But it's not that big. Warlock for example has 700 heal at lvl 4 with the talent.

File: game/scripts/npc/heroes/treant.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"DOTAHeroes"
+{
+  //=================================================================================================================
+  // HERO: Treant Protector
+  //=================================================================================================================
+  "npc_dota_hero_treant"
+  {
+    "Ability10"                                           "special_bonus_unique_treant_9"
```
Comment (DarkoniusXNG) [inline game/scripts/npc/heroes/treant.txt:8]: I am not a fan of having 2 talents upgrading the same ability at the same level. In this case 2 Nature's Grasp talents.

**Comment by TheUnholyWrath** [comment]
I didn't notice when I commented that it changed to 25 26 27 28 lmao wtf github, when I try to edit it goes back to 25 20 15 10 

1) It worked in my mode
2) & 3) I'll reshuffle the talents and change the abilities where needed. I thought grimstroke had two level 10 talents for Ink Swell but I'm just wrong. 
4) & 5) With a buffed and more usable Leech Seed slow it could be too annoying, visually it might also get a bit weird, I'll test it out.

**Comment by TheUnholyWrath** [comment]
This is the altered talent tree:

25 450 AoE Living Armor | +350 Eyes in the Forest + Overgrowth AoE
20 +80 Leech Seed Damage | + 600 Living Armor Heal
15 +75 DPS Overgrowth | +20% Leech Seed Movement Slow
10 -5s Nature's Grasp Cooldown | +4 Living Armor Bonus Armor

Other changes: 
- Eyes in the forest charge cd 60s -> 70s
- Leech Seed Cast Range and Duration scaling (400 - 900) (5s - 7.5s)
- Living Armor heal reduced to 150/300/450/600/1200/2400
- Nature's Grasp Damage increased to 30 60 90 120 240 480

A lot of these changes should help treant in the early - midgame so he's less reliant on eyes spam at that time, also 25 talent helps it scale so I wanted to slightly nerf the charge cd time (its still better than vanilla).

The extra 5% slow (on the 15 talent) might be unnecessary on Leech Seed now that its more usable.

Living Armor.

Nature's Grasp is a weird spell, I'd like to remove the 50% creep damage reduction and fiddle with the numbers a little more but it looks like its hardcoded, should be fine as is even if its double vanilla level 4 values.

**Review by DarkoniusXNG — APPROVED** [review]

**Pull request merged**